File: proyecto_santi/lib/models/departamento.dart
class Departamento {
  final int id;
  final String? codigo;
  final String nombre;
  final String? descripcion;

  Departamento({
    required this.id,
    this.codigo,
    required this.nombre,
    this.descripcion,
  });

  factory Departamento.fromJson(Map<String, dynamic> json) {
    return Departamento(
      id: json['id'] ?? json['Id'],
      codigo: json['codigo'],
      nombre: json['nombre'] ?? json['Nombre'],
      descripcion: json['descripcion'] ?? json['Descripcion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'codigo': codigo,
      'nombre': nombre,
      'descripcion': descripcion,
    };
  }
}

File: ACEXAPI/Models/Departamento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ACEXAPI.Models;

[Table("departamentos")]
public class Departamento
{
    [Key]
    [Column("id_depar")]
    public int Id { get; set; }

    [MaxLength(3)]
    [Column("codigo")]
    public string? Codigo { get; set; }

    [Required]
    [MaxLength(200)]
    [Column("nombre")]
    public string Nombre { get; set; } = string.Empty;

    [MaxLength(500)]
    public string? Descripcion { get; set; }

    // Relaciones
    public ICollection<Profesor> Profesores { get; set; } = new List<Profesor>();
    // Actividades ya NO tienen relación directa con Departamento
    // La relación es: Actividad -> Responsable (Profesor) -> Departamento
}
